packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,710 @@
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Job Run Policy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:StartJobRun',
+              'emr-containers:ListJobRuns',
+              'emr-containers:DescribeJobRun',
+              'emr-containers:CancelJobRun',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Monitoring Policy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:DescribeJobRun',
+              'elasticmapreduce:CreatePersistentAppUI',
+              'elasticmapreduce:DescribePersistentAppUI',
+              'elasticmapreduce:GetPersistentAppUIPresignedURL',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              's3:GetObject',
+              's3:ListBucket',
+            ],
+          }),
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'logs:Get*',
+              'logs:DescribeLogGroups',
+              'logs:DescribeLogStreams',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    this.updateRoleTrustPolicy(jobExecutionRole.roleName);
+
+    return jobExecutionRole;
+  }
+
+  /**
+   * Creates custom resources that call describeJobRun and executes an lambda
+   * that calls update-role-trust-policy using the inputs.
+   *
+   * @param roleName - provided from automatically generated name
+   */
+  private updateRoleTrustPolicy(roleName: string) {
+
+    // first create a custom resource to retrieve the eks namespace and eks cluster output from describe virtual cluster
+    const descVirtClust = new cr.AwsCustomResource(this, 'EMR Containers DescribeVirtualCluster SDK caller', {
+      onCreate: {
+        service: 'EMRcontainers',
+        action: 'describeVirtualCluster',
+        parameters: {
+          id: this.props.virtualClusterId.value,
+        },
+        outputPaths: ['virtualCluster.containerProvider.info.eksInfo.namespace', 'virtualCluster.containerProvider.id'],
+        physicalResourceId: cr.PhysicalResourceId.of('id'),
+      },
+      policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+        resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // next use the awscli within the lambda layer to call update-role-trust-policy using the eks namespace and eks clusterId
+    const cliLayer = new awscli.AwsCliLayer(this, 'awsclilayer');
+    const shellCliLambda = new lambda.SingletonFunction(this, 'Call Update-Role-Trust-Policy', {
+      uuid: this.renderSingletonUuid(roleName), // ensures a random uuid for a singleton function
```

Comment:
shouldn't need to include roleName here, just the literal UUID

Reply:
RoleName should be unique f right? Adding it to the uuid ensures that it's unique, the render function I have below merely concatenates the two together.

Reply:
but then we create a new function for every role, which kind of defeats the purpose of having a singleton function